scripts/merge_changelog_entries.py

```diff
@@ -0,0 +1,363 @@
+                entries[KEY_THANKS].append(stripped_line + "\n")
+                entries_processed = True
+
+    if not entries_processed:
+        print(
+            f'No entry of the format "* #NNNN <description>" found in file {file_path}'
+        )
+
+
+# process the directory in the .unreleased folder
+def process_unreleased_folders(entries):
+    # get the path to the folder inside the '.unreleased' folder
+    base_directory = os.path.join(TS_PATH, DIR_CHANGE_LOG_ENTRIES)
+    dir_list = [DIR_NEW_FEATURES, DIR_BUG_FIXES]
+    for dir_name in dir_list:
+        directory_path = os.path.join(base_directory, dir_name)
+        if not os.path.isdir(directory_path):
+            print(f"Cannot not find dir: {directory_path}")
+            sys.exit(1)
+        # walk through all the files in the folder
+        for root, _, files in os.walk(directory_path):
+            for file in files:
+                if file == TEMPLATE_FILE_NAME:
+                    continue
+                file_path = os.path.join(root, file)
+                process_changelog_file(dir_name, file_path, entries)
+
+
+def process_unreleased_section(lines, start_index, entries):
+    unreleased_features = unreleased_bugfixes = unreleased_thanks = False
+    for i in range(start_index, len(lines)):
+        line = lines[i]
+        if line.strip() == "":
+            continue
+        # unreleased section ends at the start of a new version
+        if re.search(VERSION_PREFIX + VERSION_PATTERN, line):
+            break
+        patterns = [HEAD_FEATURES, HEAD_BUG_FIXES, HEAD_THANKS]
+        found = [re.search(re.escape(pat), line, re.IGNORECASE) for pat in patterns]
+        # found a header, note and proceed to next line
+        if any(found):
+            unreleased_features, unreleased_bugfixes, unreleased_thanks = [
+                bool(match) for match in found
+            ]
+            continue
+        if check_line_format(PATTERN_ENTRY, line):
+            if unreleased_features:
+                entries[KEY_FEATURES].append(line)
+            elif unreleased_bugfixes:
+                entries[KEY_BUGFIXES].append(line)
+        elif check_line_format(PATTERN_THANKS_ENTRY, line) and unreleased_thanks:
+            entries[KEY_THANKS].append(line)
+        else:
+            # if not version or a PR/thanks entry then it is header/footer
+            if unreleased_features or unreleased_bugfixes or unreleased_thanks:
+                entries[KEY_FOOTER].append(line)
+            else:
+                info_line = re.sub(VERSION_PATTERN, "{version}", line)
+                info_regex = re.compile(info_line.strip(), flags=re.MULTILINE)
+                # skip the INFO_STRING from the existing file
+                if info_regex.search(INFO_STRING):
+                    continue
+                entries[KEY_HEADER].append(line)
+
+
+# read the entries in the "##unreleased" section if any
+def read_change_log(lines, entries, new_version):
```

Comment:
I think we can assume that this section will go away if we are switching to this approach, which means that there is no need to handle it, nor rely on it being present.